lib/presentation/widget/shared_space/file_surfing/workgroup_nodes_surfing_viewmodel.dart

```diff
@@ -630,4 +635,84 @@ class WorkGroupNodesSurfingViewModel extends BaseViewModel {
+    _appNavigation.popBack();
+    store.dispatch(DisableUploadButtonAction());
+
+    final nodeName = workGroupNode is WorkGroupDocument ? AppLocalizations.of(context).file : AppLocalizations.of(context).folder;
+
+    EditTextModalSheetBuilder()
+        .key(Key('rename_work_group_node_modal'))
+        .title(AppLocalizations.of(context).rename_node(nodeName.toLowerCase()))
+        .cancelText(AppLocalizations.of(context).cancel)
+        .onConfirmAction(AppLocalizations.of(context).rename,
+            (value) => store.dispatch(_renameWorkGroupNodeAction(context, value, workGroupNode)))
+        .onCancelAction(() => store.dispatch(EnableUploadButtonAction()))
+        .setErrorString((value) => _getErrorString(context, workGroupNode, value))
+        .setTextController(
+          TextEditingController.fromValue(
+            TextEditingValue(
+                text: workGroupNode.name,
+                selection: TextSelection(
+                    baseOffset: 0,
+                    extentOffset: workGroupNode.name.length
+                )
+            ),
+          ))
+        .show(context);
+  }
+
+  String _getErrorString(BuildContext context, WorkGroupNode workGroupNode, String value) {
+    final listName = workGroupNode is WorkGroupDocument
+        ? _workGroupNodesList
+            .whereType<WorkGroupDocument>()
+            .map((node) => node.name)
+            .toList()
+        : _workGroupNodesList
+            .whereType<WorkGroupFolder>()
+            .map((node) => node.name)
+            .toList();
+
+    return _verifyNameInteractor
+        .execute(
+          value,
+          [
+            EmptyNameValidator(),
+            DuplicateNameValidator(listName),
+            SpecialCharacterValidator(),
+            if (workGroupNode is WorkGroupDocument) LastDotValidator()
+          ])
+        .fold((failure) {
+          if (failure is VerifyNameFailure) {
+            final nodeName = workGroupNode is WorkGroupDocument ? AppLocalizations.of(context).file : AppLocalizations.of(context).folder;
+            if (failure.exception is EmptyNameException) {
+              return AppLocalizations.of(context).node_name_not_empty(nodeName);
+            } else if (failure.exception is DuplicatedNameException) {
+              return AppLocalizations.of(context).node_name_already_exists(nodeName);
+            } else if (failure.exception is SpecialCharacterException) {
+              return AppLocalizations.of(context).node_name_contain_special_character(nodeName);
+            } else if (failure.exception is LastDotException) {
+              return AppLocalizations.of(context).node_name_contain_last_dot(nodeName);
+            } else {
+              return null;
+            }
+          } else {
+            return null;
+          }
+        }, (success) => null);
+  }
+
+  OnlineThunkAction _renameWorkGroupNodeAction(BuildContext context, String newName, WorkGroupNode workGroupNode) {
+    Navigator.pop(context);
```

Comment:
Why don't use _appNavigation.popBack();?

Reply:
This error I explained in A Dat's comment. I am rebuilding the structure of the shared space to solve this problem.